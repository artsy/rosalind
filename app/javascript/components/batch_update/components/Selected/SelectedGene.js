import React from 'react'
import SelectedComponent from './SelectedComponent'

const SelectedGene = props => {
  return <SelectedComponent name={props.name} onRemove={props.onRemove} />
}

export { SelectedGene }
