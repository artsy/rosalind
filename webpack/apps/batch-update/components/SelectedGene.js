import React from 'react'

class SelectedGene extends React.Component {
  constructor (props) {
    super(props)
    this.handleRemove = this.handleRemove.bind(this)
  }

  handleRemove (e) {
    e.preventDefault()
    this.props.onRemoveGene(this.props.name)
  }

  render () {
    const { className, name } = this.props
    return (
      <div className={className}>
        {name}
        <a href='#' className='remove'
          onClick={this.handleRemove}>âœ•</a>
      </div>
    )
  }
}

SelectedGene.propTypes = {
  name: React.PropTypes.string.isRequired,
  onRemoveGene: React.PropTypes.func.isRequired
}

/* default styled component */

import styled from 'styled-components'

const StyledSelectedGene = styled(SelectedGene)`
  font-weight: bold;
  margin: 0.25em 0;

  a.remove {
    display: inline-block;
    height: 2em;
    width: 2em;
    line-height: 2em;
    text-align: center;
    color: #ddd;
    margin-left: 0.5em;
    border-radius: 1em;
    text-decoration: none;
    &:hover {
      color: #333;
      cursor: pointer;
      background: #eee;
    }
  }
`

export default StyledSelectedGene
