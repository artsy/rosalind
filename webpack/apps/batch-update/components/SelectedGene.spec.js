import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import SelectedGene from './SelectedGene'

let props

beforeEach(() => {
  props = {
    name: 'Kawaii',
    onRemoveGene: jest.fn()
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SelectedGene {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('fires the remove handler on click', () => {
  const wrapper = mount(<SelectedGene {...props} />)
  const mockClickEvent = { preventDefault: jest.fn() }
  wrapper.find('a').simulate('click', mockClickEvent)
  expect(props.onRemoveGene.mock.calls.length).toEqual(1)
})
