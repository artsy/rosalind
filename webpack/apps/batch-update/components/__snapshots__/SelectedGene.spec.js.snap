exports[`test renders correctly 1`] = `
<div
  className="crrNnH">
  Kawaii
  <a
    className="remove"
    href="#"
    onClick={[Function]}>
    âœ•
  </a>
</div>
`;
